src/gl/gl_builders.js

```diff
@@ -145,213 +145,149 @@ GLBuilders.buildPolylines = function (
-        var line_debug = null;
-        if (intersection != null) {
-            var intersect_outer = intersection;
-
-            // Cap the intersection point to a reasonable distance (as join angle becomes sharper, miter joint distance would approach infinity)
-            var len_sq = Vector.lengthSq([intersect_outer[0] - joint[0], intersect_outer[1] - joint[1]]);
-            var miter_len_max = 3; // multiplier on line width for max distance miter join can be from joint
-            if (len_sq > (width * width * miter_len_max * miter_len_max)) {
-                line_debug = 'distance';
-                intersect_outer = Vector.normalize([intersect_outer[0] - joint[0], intersect_outer[1] - joint[1]]);
-                intersect_outer = [
-                    joint[0] + intersect_outer[0] * miter_len_max,
-                    joint[1] + intersect_outer[1] * miter_len_max
-                ];
-            }
+        // Do this with the rest (except the last one)
+        for(i = 1; i < lineSize - 1 ; i++){
+            prevCoord = currCoord;
+            currCoord = nextCoord;
+            nextCoord = line[i+1];
 
-            var intersect_inner = [
-                (joint[0] - intersect_outer[0]) + joint[0],
-                (joint[1] - intersect_outer[1]) + joint[1]
-            ];
+            //  a) Store prev normal and compute the next one
+            normPrevCurr = normCurrNext;
+            normCurrNext = Vector.normalize( Vector.perp( currCoord, nextCoord ) );
+            rightNorm = Vector.normalize( Vector.add(normPrevCurr, normCurrNext) );
 
-            vertices.push(
-                intersect_inner, intersect_outer, pa_inner[0],
-                pa_inner[0], intersect_outer, pa_outer[0],
+            var scale = Math.sqrt(2 / (1 + Vector.dot(normPrevCurr,normCurrNext)));
+            rightNorm = Vector.mult(rightNorm,scale);
+            //  TODO:
+            //          - compare to the miterlimit
+            //          - if bigger 
 
-                pb_inner[1], pb_outer[1], intersect_inner,
-                intersect_inner, pb_outer[1], intersect_outer
-            );
+            addVertexPair(currCoord, rightNorm, i/lineSize );
         }
-        else {
-            // Line segments are parallel, use the first outer line segment as join instead
-            line_debug = 'parallel';
-            pa_inner[1] = pb_inner[0];
-            pa_outer[1] = pb_outer[0];
-
-            vertices.push(
-                pa_inner[1], pa_outer[1], pa_inner[0],
-                pa_inner[0], pa_outer[1], pa_outer[0],
-
-                pb_inner[1], pb_outer[1], pb_inner[0],
-                pb_inner[0], pb_outer[1], pb_outer[0]
-            );
+
+        rightNorm = Vector.normalize( normCurrNext );
+        addVertexPair(nextCoord, rightNorm, 1 );
+
+        // Add vertices to buffer acording their index
+        for (i = 0; i < lineSize - 1; i++) {
```

Comment:
I noticed that you reuse the `i` variable in this function. Feel free to use `let` instead of `var`. `Let` respects block scoping while `var` only understands function scope. 

https://github.com/google/traceur-compiler/wiki/LanguageFeatures#block-scoped-binding
I believe this is turned on by default now in traceur